frontend_app_gallery/src/component/MainPreview.jsx

```diff
@@ -32,28 +40,40 @@ class MainPreview extends React.Component {
     return (
       <div className='carousel__item__preview'>
         <span className='carousel__item__preview__content'>
-          {!state.imageLoaded && (
+          {state.imageLoaded === IMG_LOAD_STATE.LOADING && (
             <div className='gallery__loader'>
               <i className='fa fa-spinner fa-spin gallery__loader__icon' />
             </div>
           )}
-          <div className='carousel__item__preview__content__image'>
-            <img
-              src={props.previewSrc}
-              className={classnames(`rotate${props.rotationAngle}`, state.imageLoaded ? 'img-thumbnail' : null)}
-              onClick={props.handleClickShowImageRaw}
-              onLoad={this.onLoad.bind(this)}
-              width={width && state.width > width ? width : null}
-              alt={props.fileName}
-            />
-          </div>
+          {state.imageLoaded !== IMG_LOAD_STATE.ERROR
+            ? (
+              <div className='carousel__item__preview__content__image'>
+                <img
+                  src={props.previewSrc}
+                  className={classnames(`rotate${props.rotationAngle}`, state.imageLoaded ? 'img-thumbnail' : null)}
+                  onClick={props.handleClickShowImageRaw}
+                  onLoad={this.onLoad}
+                  onError={this.onError}
+                  width={width && state.width > width ? width : null}
+                  alt={props.fileName}
+                />
+              </div>
+            ) : (
+              <div className='carousel__item__preview__error'>
+                <div className='carousel__item__preview__error__message'>
+                  <i className='fa fa-fw fa-exclamation-triangle carousel__item__preview__error__icon' />
+                  <div>{props.t('No preview available')}</div>
```

Comment:
you added a translation key but I don't see any addition into translation files.
Did you forget to run the translation script ?

Reply:
Yes I forgot, I didn't noticed because the key is already in app_file